http/client.go

```diff
@@ -68,6 +73,16 @@ func NewClient() Clienter {
 	return &newClient
 }
 
+func NewClientWithAwsSigner(awsFilename, awsProfile, awsRegion, awsService string) Clienter {
```

Comment:
Why return an interface here? Generally it's better to return structs unless there's a good reason

Reply:
Again, this is an idiomatic go where the initialisers will return an interface not the underlying struct objects. Actually if you try to return an underlying struct the linter (gocritic) I think will complain saying the return type should be an interface.

Reply:
https://www.ardanlabs.com/blog/2016/10/avoid-interface-pollution.html

Reply:
added comment below